site/app/libraries/routers/WebRouter.php

```diff
@@ -94,10 +94,10 @@ private function loginCheck() {
             $old_request_url = $this->request->getUriForPath($this->request->getPathInfo());
             $this->request = Request::create(
                 '/authentication/login',
-                'GET'
+                'GET',
+                ['old' => base64_encode($old_request_url)]
```

Comment:
I'm not too familiar with this code but will this 'old' url get passed through PHP if so should it be URL-encoded instead of base64.
Since base64 allows the '+', '=', and '/' chars which are reserved in ULRs I wasn't sure if allowing them might break the URL being built here.

Reply:
@shailpatels 

Thanks a lot for mentioning this. I used `urlencode` in the beginning, but it weirdly failed to work (I don't know why, probably because the old way of handling it has some inherent problem). Now I tried it again and it works in my local machine now (please help me check that. Thanks!). Yes, `urlencode` and `urldecode` are more suitable in this scenario and using them is a more common practice.